streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -233,26 +280,28 @@ public void initializeTopology() {
      * - resume the task
      * </pre>
      */
-    @Override
     public void resume() {
         log.debug("Resuming");
-        initializeMetadata();
+        if (state() == State.SUSPENDED) {
+            initializeMetadata();
+            transitionTo(State.RUNNING);
+        }
```

Comment:
Ditto here, we if state is not SUSPENDED it should be a bug, and we can throw IllegalState here.